corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -234,10 +234,14 @@ hqDefine('app_manager/js/app_manager', function () {
     var _initMenuItemSorting = function () {
         nestChildModules();
         initDragHandles();
-        $('.sortable').each(function () {
-            initSortable($(this));
-        });
-        $('.sort-action').hide();
+        if (modulesWereReordered()) {
+            promptToSaveOrdering();
+        } else {
+            $('.sortable').each(function () {
+                initSortable($(this));
+            });
+            $('.sort-action').hide();
+        }
```

Comment:
The promptToSaveOrdering will be dismissed if the user clicks outside the modal, but they won't be able to reorder the modules, since they're not initialized here.

As per the commit message, I don't expect this situation to arise often, but it could, since we currently have applications with child modules that don't follow their parent.

Reply:
I think this is a good ui compromise.